bin/prover-client/src/prover.rs

```diff
@@ -73,8 +73,8 @@ impl ProverState {
         }
     }
 
-    fn get_prover_status(&self, task_id: Uuid) -> Option<&ProvingTaskState> {
-        self.tasks_status.get(&task_id)
+    fn get_prover_status(&mut self, task_id: &Uuid) -> Entry<Uuid, ProvingTaskState> {
+        self.tasks_status.entry(*task_id)
```

Comment:
Is it ok to leak this impl detail?  I suppose it is since it's not `pub`.

Reply:
I suppose yes, it is ok, exactly for the reason of not being `pub`